### ISSUE TITLE
.NET UTF-7 encoding obsoletion

### ISSUE THREAD

**Issue by xtqqczze** [PowerShell/PowerShell#13358]
According to <https://github.com/dotnet/designs/blob/master/accepted/2020/better-obsoletion/obsoletions-in-net5.md>, .NET 5 will obsolete the UTF-7 encoding:

>  The UTF-7 encoding is insecure and should not be used. Consider using UTF-8 instead.

**Comment by jborean93** [comment]
I'm curious, is UTF-7 being used anywhere in PowerShell? I've certainly never come across it in the wild.

**Comment by xtqqczze** [comment]
> 
> 
> I'm curious, is UTF-7 being used anywhere in PowerShell? I've certainly never come across it in the wild.

It is exposed in public APIs such as `Get-Content -Encoding UTF7`

**Comment by iSazonov** [comment]
Yes, it is public and removing it will be a breaking change. But if .Net 5.0 will deprecate this we will have to do the same.

/cc @SteveL-MSFT @daxian-dbw It is for PowerShell Committee review.

**Comment by xtqqczze** [comment]
@iSazonov `System.Text.Encoding.UTF7` and `System.Text.UTF7Encoding` APIs are not being removed from .NET 5, they are being marked with `ObsoleteAttribute`. So we have a choice whether to remove or to suppress the warnings.

https://github.com/dotnet/designs/blob/master/accepted/2020/better-obsoletion/better-obsoletion.md

**Comment by iSazonov** [comment]
Search on GiHub https://github.com/search?l=PowerShell&q=UTF7&type=Code returns ~1K results. It is very small and it seems they simply replicate PowerShell support. I'd speculate that nobody uses UTF7 and we can remove it at all. The same for C#.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee discussed this, we have code that handles `Encoding` passed to cmdlets that accept that type for parameters.  We agreed to add a warning message if the user specifies "UTF-7" as part of type conversion to warn the user that this encoding is obsolete, but we should not remove it while .NET itself still implements it even though obsolete.

**Comment by xtqqczze** [comment]
See also: #13484
